import package naming conventions (#4851)
ros2/ros2_documentation: source/The-ROS2-Project/Contributing/Developer-Guide.rst

The following sections are the specific development rules we follow to ensure core packages are of the highest quality ('Level 1').
We recommend all ROS developers strive to adhere to the following policies to ensure quality across the ROS ecosystem.

> For more specific code recommendations please see :doc:`the Quality Guide <Quality-Guide>`

Reviewer: ```suggestion
For more specific code recommendations please see :doc:`the Quality Guide <Quality-Guide>`.
```

* For compiled code, the ABI is considered part of the public interface.
  Any change that requires recompiling dependent code is considered major (breaking).

>   * ABI breaking changegit@github.com:ros2/ros2_documentation.gits *can* be made in a minor version bump *before* a distribution release (getting added to the rolling release).

Reviewer: This looks like an accidental middle mouse button (paste) click!

Author: ```suggestion
  * ABI breaking changes *can* be made in a minor version bump *before* a distribution release (getting added to the rolling release).
```

Woops yes

We will update the status (if you don't have the permission) and you can start working on a pull request.
If you contribute regularly we will likely just grant you permission to manage the labels etc. yourself.

> Package Naming Conventions

Reviewer: This is part of the "contributing" guide (i.e., core-y stuff) while the original ROS wiki page is more of a general naming guidelines page (i.e., end users/devs stuff). Should we instead move this section to a more general page and include a link here? Just a minor thought.

Author: Yeah, the scope of this document is a bit mixed. We probably want to split it out to general developer guide, And a second part for core team developer guides. For now I mostly wanted to capture the content, and I couldn't find a better place than the developer guide.

Reviewer: Alright, we can always re-structure/move sections later.

We will update the status (if you don't have the permission) and you can start working on a pull request.
If you contribute regularly we will likely just grant you permission to manage the labels etc. yourself.

Package Naming Conventions
^^^^^^^^^^^^^^^^^^^^^^^^^^

Names play an important role in ROS and following naming conventions simplifies the process of learning and understanding large systems.

The ROS packages occupy a flat namespace, so naming should be done carefully and consistently. There is a standard for package naming in `REP-144 <https://www.ros.org/reps/rep-0144.html>`__

* Package names should follow common C variable naming conventions: lower case, start with a letter, use underscore separators, e.g. laser_viewer

> * Package names should be specific enough to identify what the package does. For example, a motion planner is not called planner. If it implements the wavefront propagation algorithm, it might be called wavefront_planner. There's obviously tension between making a name specific and keeping it from becoming overly verbose.

Reviewer: Would you mind changing this list to one sentence per line? Looks like we do it even for each list item, e.g., https://github.com/ros2/ros2_documentation/blob/d019184fd36b8272f4c1e03deaebf2bd67c25b9e/source/The-ROS2-Project/Contributing/Developer-Guide.rst?plain=1#L165-L168

>   * Using catchall names such as utils should be avoided as they do not scope what goes into the package or what should be outside the package.

Reviewer: This item doesn't get indented correctly. Looks like this list needs empty lines in between all items?

Author: fixed

> * To check whether a name is taken, consult `<index.ros.org/packages/>`_. If you'd like your repository included in that list, see the `rosdistro Contributing Guide <https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md>`__.

Reviewer: The first link is broken, it probably just needs an extra underscore:

```suggestion
* To check whether a name is taken, consult `<index.ros.org/packages/>`__. If you'd like your repository included in that list, see the `rosdistro Contributing Guide <https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md>`__.
```

Might be worth adding `https://` to the link too.

If you contribute regularly we will likely just grant you permission to manage the labels etc. yourself.

Package Naming Conventions
^^^^^^^^^^^^^^^^^^^^^^^^^^

Names play an important role in ROS and following naming conventions simplifies the process of learning and understanding large systems.

The ROS packages occupy a flat namespace, so naming should be done carefully and consistently. There is a standard for package naming in `REP-144 <https://www.ros.org/reps/rep-0144.html>`__

* Package names should follow common C variable naming conventions: lower case, start with a letter, use underscore separators, e.g. laser_viewer

* Package names should be specific enough to identify what the package does.
  For example, a motion planner is not called planner.
  If it implements the wavefront propagation algorithm, it might be called wavefront_planner.
  There's obviously tension between making a name specific and keeping it from becoming overly verbose.

  * Using catchall names such as utils should be avoided as they do not scope what goes into the package or what should be outside the package.

> * To check whether a name is taken, consult `<index.ros.org/packages/>`__.

Reviewer: This link doesn't work. It leads to `/The-ROS2-Project/Contributing/index.ros.org/packages/`. I assume this should fix it:

```suggestion
* To check whether a name is taken, consult `<https://index.ros.org/packages/>`__.
```
